bigbluebutton-html5/imports/ui/services/audio-manager/index.js

```diff
@@ -128,7 +128,9 @@ class AudioManager {
   }
 
   toggleMuteMicrophone() {
-    makeCall('toggleSelfVoice');
+    makeCall('toggleSelfVoice').then(() => {
+      this.onToggleMicrophoneMute();
```

Comment:
There is no guarantee that the toggle actually happened and this could cause problems, the only source of truth should be mongo/redis.

Eg.
 - makeCall('toggleSelfVoice')
 - server send redis event
 - something bad happen on akka or w/e
 - onToggleMicrophoneMute()
 - now akka and audio-manager have different states, since one failed silently

Reply:
I agree. Observer is probably better. The thing is the current observer on VoiceUsers never triggers. I'll look into it further.